python-threatexchange/threatexchange/cli/dataset/simple_serialization.py

```diff
@@ -87,3 +90,61 @@ def load(
                         )
                     )
         return ret
+
+
+class HMASerialization(CliIndicatorSerialization):
+    """
+    A Serialization for HMA Similar to CliIndicatorSerialization but with
+    Indicator ID instead of Descriptor ID
+    """
+
+    def __init__(
+        self,
+        indicator_id: str,
+        indicator_type: str,
+        indicator: str,
+        rollup: SimpleDescriptorRollup,
+    ):
+        self.indicator_id = indicator_id
+        self.indicator_type = indicator_type
+        self.indicator = indicator
+        self.rollup = rollup
+
+    def as_csv_row(self) -> t.Tuple:
+        """indicator details and descriptor rollup without descriptor ID"""
+        return (self.indicator_id, self.indicator) + self.rollup.as_row_without_id()
+
+    @classmethod
+    def from_threat_updates_json(cls, app_id, te_json):
+        return cls(
+            te_json["id"],
+            te_json["type"],
+            te_json["indicator"],
+            SimpleDescriptorRollup.from_threat_updates_json(app_id, te_json),
+        )
+
+    @classmethod
+    def load(
+        cls, state_dir: pathlib.Path
+    ) -> t.List["threat_updates.ThreatUpdateSerialization"]:
+        """Load this serialization from the state directory"""
+        ret = []
+        pattern = r"simple\.([^.]+)" + re.escape(Dataset.EXTENSION)
+        for path in state_dir.glob(f"simple.*{Dataset.EXTENSION}"):
+            match = re.match(pattern, path.name)
+            if not match or not path.is_file():
+                continue
+            indicator_type = match.group(1)
+            # Violate your warranty with class state! Not threadsafe!
+            csv.field_size_limit(path.stat().st_size)  # dodge field size problems
+            with path.open("r", newline="") as f:
+                for row in csv.reader(f):
+                    ret.append(
+                        cls(
+                            row[0],
+                            indicator_type,
+                            row[1],
+                            SimpleDescriptorRollup.from_row(row[2:]),
```

Comment:
Does this still work or does it expect an id?

Reply:
You're right. This is expecting a descriptor idea. Didn't realize this earlier because I wasn't actually applying updates yet so had nothing to load. I think this means we either need to store the descriptor ID or stop using SimpleDescriptorRollup. I think the former is better